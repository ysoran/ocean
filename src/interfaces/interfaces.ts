import { Dispatch, SetStateAction } from "react";

export interface IShipInterface {
  MMSI: number;
  TIME: string;
  LONGITUDE: number;
  LATITUDE: number;
  COG: number;
  SOG: number;
  HEADING: number;
  ROT: number;
  NAVSTAT: number;
  IMO: number;
  NAME: string;
  CALLSIGN: string;
  TYPE: number;
  A: number;
  B: number;
  C: number;
  D: number;
  DRAUGHT: number;
  DEST: string;
  ETA: string;
}

export interface ILister {
  ships: Array<IShipInterface> | undefined;
  showHeader: boolean;
  setShowHeader: (opt: boolean) => void;
  port: IPort;
}

export interface IPort {
  name: string;
  country_code: string;
  city_code: string;
  location: string;
  value: string;
}

export interface IMapProps {
  center?: any;
  zoom?: number;
  ships: Array<IShipInterface>;
  port: IPort;
}

export interface IPortProps {
  lat: number;
  lng: number;
  name?: string;
}

export interface IShipProps {
  lat: number;
  lng: number;
  name: string;
  idle: boolean;
  eta: string;
  dest: string;
  imo: number;
}

export interface ISearchPanel {
  handleSearch: (
    port: number,
    startDate: string,
    endDate: string,
    distance: number | string,
    includeIdleVessels: boolean,
    setSearchWarning: Dispatch<SetStateAction<string | undefined>>
  ) => void;
  clear: any;
}
