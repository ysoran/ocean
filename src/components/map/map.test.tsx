import React from 'react';
import { shallow } from 'enzyme';
import {IShipInterface,IPort} from '../../interfaces/interfaces';
import Map from './map';

const setup = () => {
    const wrapper = shallow(<Map ships={[] as Array<IShipInterface>} port={{} as IPort} />);
    return {
        wrapper,
    };
};

describe('Map component', () => {
    it('should render snapshot', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });
});
