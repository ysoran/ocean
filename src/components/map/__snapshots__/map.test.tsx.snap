// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component should render snapshot 1`] = `ShallowWrapper {}`;
